crates/ruff_cli/src/cache.rs

```diff
@@ -296,3 +296,164 @@ pub(crate) fn init(path: &Path) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use std::env::temp_dir;
+    use std::fs;
+    use std::io::{self, Write};
+    use std::path::Path;
+
+    use ruff::settings::{flags, AllSettings};
+
+    use crate::cache::{self, Cache};
+    use crate::diagnostics::{lint_path, Diagnostics};
+
+    #[test]
+    fn same_results() {
+        let mut cache_dir = temp_dir();
+        cache_dir.push("ruff_tests/cache_same_results");
+        let _ = fs::remove_dir_all(&cache_dir);
+        cache::init(&cache_dir).unwrap();
+
+        let settings = AllSettings::default();
+
+        let package_root = fs::canonicalize("../ruff/resources/test/fixtures").unwrap();
+        let cache = Cache::open(&cache_dir, package_root.to_owned(), &settings.lib);
+        assert_eq!(cache.new_files.lock().unwrap().len(), 0);
+
+        let mut paths = Vec::new();
+        let mut expected_diagnostics = Diagnostics::default();
+        for entry in fs::read_dir(&package_root).unwrap() {
+            let entry = entry.unwrap();
+            if !entry.file_type().unwrap().is_dir() {
+                continue;
+            }
+
+            for entry in fs::read_dir(entry.path()).unwrap() {
+                let entry = entry.unwrap();
+                if !entry.file_type().unwrap().is_file() {
+                    continue;
+                }
+
+                let path = entry.path();
+                expected_diagnostics += lint_path(
+                    &path,
+                    Some(&package_root),
+                    &settings,
+                    Some(&cache),
+                    flags::Noqa::Enabled,
+                    flags::FixMode::Generate,
+                )
+                .unwrap();
+                paths.push(path);
+            }
+        }
+        assert!(paths.len() >= 1, "no files checked");
```

Comment:
Could we store the values of saved files in the cache and use it in the assertion on line 357?

Reply:
I've added commit 42f2bf17421d0086bc86e84c7c0ec562b232ce15 that checks the cached files a little more strictly, though it's a little brittle around dealing with parse errors (which we don't cache).